Error: io-dev-functions

After install i got this

]  ERROR  Cannot start nuxt:  Cannot find module '/PROJECT/.nuxt/io-dev-functions' imported from /PROJECT/node_modules/nuxt3-socket.io/dist/module.mjs
[0] 
[0]   at new NodeError (node:internal/errors:387:5)
[0]   at finalizeResolution (node:internal/modules/esm/resolve:429:11)
[0]   at moduleResolve (node:internal/modules/esm/resolve:1006:10)
[0]   at defaultResolve (node:internal/modules/esm/resolve:1214:11)
[0]   at nextResolve (node:internal/modules/esm/loader:165:28)
[0]   at ESMLoader.resolve (node:internal/modules/esm/loader:844:30)
[0]   at ESMLoader.getModuleJob (node:internal/modules/esm/loader:431:18)
[0]   at ESMLoader.import (node:internal/modules/esm/loader:528:22)
[0]   at importModuleDynamically (node:internal/modules/esm/translators:110:35)
[0]   at importModuleDynamicallyCallback (node:internal/process/esm_loader:35:14)

I got the same error!

```
 ERROR  Cannot start nuxt:  Cannot find module '/Volumes/Projects/fetvguide-website/.nuxt/io-dev-functions' imported from /Volumes/Projects/fetvguide-website/node_modules/.pnpm/nuxt3-socket.io@0.0.11/node_modules/nuxt3-socket.io/dist/module.mjs

  at new NodeError (node:internal/errors:400:5)
  at finalizeResolution (node:internal/modules/esm/resolve:260:11)
  at moduleResolve (node:internal/modules/esm/resolve:879:10)
  at defaultResolve (node:internal/modules/esm/resolve:1087:11)
  at nextResolve (node:internal/modules/esm/loader:161:28)
  at ESMLoader.resolve (node:internal/modules/esm/loader:834:30)
  at ESMLoader.getModuleJob (node:internal/modules/esm/loader:415:18)
  at ESMLoader.import (node:internal/modules/esm/loader:516:22)
  at importModuleDynamically (node:internal/modules/esm/translators:108:35)
  at importModuleDynamicallyCallback (node:internal/process/esm_loader:36:14)
```

--- a/src/module.ts
+++ b/src/module.ts
@@ -40,26 +40,31 @@ export default defineNuxtModule<ModuleOptions>({
     await scanRemoteFunctions()
 
     addTemplate({
-      filename: 'io-dev-functions.ts',
+      filename: 'io-dev-functions.mjs',
       write: true,
       getContents () {
-        return `
-          ${files.map((file, index) => `import function${index} from '${file.replace('.ts', '')}'`).join('\n')}
+         return `
+          import jiti from 'jiti';
+          const _require = jiti(process.cwd(), { interopDefault: true, esmResolve: true });
+
+          ${files.map((file, index) => `const function${index} = _require('${file.replace(".ts", "")}');`).join("\n")}
           export {
-            ${files.map((_, index) => `function${index}`).join(',\n')}
+            ${files.map((_, index) => `function${index}`).join(",\n")}
           }
-        `
+        `;
       }
     })
 
     if (nuxt.options.dev) {
-      const devFunctionsPath = resolve(nuxt.options.buildDir, 'io-dev-functions')
+      const devFunctionsPath = resolve(nuxt.options.buildDir, 'io-dev-functions.mjs');
 
-      nuxt.hook('listen', async (httpServer) => {
-        const io = new SocketServer(httpServer, options.serverOptions)
-        const functions = await import(devFunctionsPath)
-        Object.keys(functions).forEach((fn) => {
-          functions[fn](io)
+      nuxt.hook('listen', (httpServer) => {
+        nuxt.hook("app:templatesGenerated", async () => {
+          const io = new SocketServer(httpServer, options.serverOptions)
+          const functions = await import(devFunctionsPath)
+          Object.keys(functions).forEach((fn) => {
+            functions[fn](io)
+          })
         })
       })
     }

